src/System/Wlog/LoggerConfig.hs

```diff
@@ -241,6 +244,22 @@ data LoggerConfig = LoggerConfig
 
 makeLenses ''LoggerConfig
 
+instance Semigroup LoggerConfig where
+    lc1 <> lc2 = LoggerConfig
+        { _lcRotation        = orCombiner  _lcRotation
+        , _lcTermSeverityOut = orCombiner  _lcTermSeverityOut
+        , _lcTermSeverityErr = orCombiner  _lcTermSeverityErr
+        , _lcShowTime        = andCombiner _lcShowTime
+        , _lcShowTid         = andCombiner _lcShowTid
+        , _lcConsoleAction   = andCombiner _lcConsoleAction
+        , _lcMapper          = andCombiner _lcMapper
+        , _lcTree            = andCombiner _lcTree
+        }
+      where
+        orCombiner  field = field lc1 <|> field lc2
+        andCombiner field = field lc1  <> field lc2
```

Comment:
Could you please move these two functions at top level and use them for `LoggerTree` `Semigroup` instance implementation? I propose to have names `appendBy` (or `andBy`) and `altBy` (or `orBy`). You can still have functions inside `where` though as partial specialisations of those two.

Reply:
Though... I just thought that you can just duplicate code. It will be shorter.